fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/api/LoanTransactionsApiResource.java

```diff
@@ -439,6 +477,59 @@ private String retrieveTransaction(final Long loanId, final String loanExternalI
         return this.toApiJsonSerializer.serialize(settings, transactionData, this.responseDataParameters);
     }
 
+    private Page<LoanTransactionData> retrieveTransactions(final Long loanId, final String loanExternalIdStr, final String excludedTypesStr,
+            Pageable pageable, final UriInfo uriInfo) {
+
+        this.context.authenticatedUser().validateHasReadPermission(RESOURCE_NAME_FOR_PERMISSIONS);
+
+        ExternalId loanExternalId = ExternalIdFactory.produce(loanExternalIdStr);
+        Long resolvedLoanId = getResolvedLoanId(loanId, loanExternalId);
+
+        Set<LoanTransactionType> excludedTransactionTypes = new HashSet<>();
+        if (StringUtils.isNotBlank(excludedTypesStr)) {
+            excludedTransactionTypes = new HashSet<>(
+                    Arrays.stream(excludedTypesStr.split("\\s*,\\s*")).map(this::transactionTypeFromParam).toList());
+        }
+
+        return loanReadPlatformService.retrieveLoanTransactions(resolvedLoanId, excludedTransactionTypes, pageable);
+    }
+
+    private LoanTransactionType transactionTypeFromParam(String transactionTypeParam) {
+        return switch (transactionTypeParam) {
+            case DataTableApiConstant.disbursement -> LoanTransactionType.DISBURSEMENT;
```

Comment:
Datatableapi?

Reply:
I saw that there were already some loan related constants in it, but looking at it now it does look strange to put them there. Moved the new ones to their own place.